protocol/invoker.go

```diff
@@ -88,11 +88,12 @@ func (bi *BaseInvoker) Invoke(context context.Context, invocation Invocation) Re
 	return &RPCResult{}
 }
 
-// Destroy changes available and destroyed flag
+// Destroy changes available and destroyed flag and release the url's allocated memory
 func (bi *BaseInvoker) Destroy() {
 	logger.Infof("Destroy invoker: %s", bi.GetURL())
 	bi.destroyed.Store(true)
 	bi.available.Store(false)
+	bi.url = nil
```

Comment:
If set nil, is there any risk of a null pointer elsewhere

Reply:
As u can see in other changed files, in other places where`url` are used, I changed the order, that is, destroyed them in the end to avoid null pointer issues; But still requires careful review;